inference.py

```diff
@@ -412,34 +542,37 @@ def main(params: dict):
         else:
             model, _ = load_from_checkpoint(loaded_checkpoint, model)
         # LOOP THROUGH LIST OF INPUT IMAGES
-        for info in tqdm(list_img, desc='Inferring from images', position=0):
-            img_name = Path(info['tif']).name
-            local_gpkg = Path(info['gpkg']) if 'gpkg' in info.keys() and info['gpkg'] else None
-            gpkg_name = local_gpkg.stem if local_gpkg else None
-            if bucket:
-                local_img = f"Images/{img_name}"
-                bucket.download_file(info['tif'], local_img)
-                inference_image = f"Classified_Images/{img_name.split('.')[0]}_inference.tif"
-                if info['meta']:
-                    if info['meta'] not in bucket_file_cache:
-                        bucket_file_cache.append(info['meta'])
-                        bucket.download_file(info['meta'], info['meta'].split('/')[-1])
-                    info['meta'] = info['meta'].split('/')[-1]
-            else:  # FIXME: else statement should support img['meta'] integration as well.
-                local_img = Path(info['tif'])
-                Path.mkdir(working_folder.joinpath(local_img.parent.name), parents=True, exist_ok=True)
-                inference_image = working_folder.joinpath(local_img.parent.name,
-                                                          f"{img_name.split('.')[0]}_inference.tif")
-            with rasterio.open(local_img, 'r') as raster:
-                logging.info(f'Reading image as array: {raster.name}')
-                img_array, raster, _ = image_reader_as_array(input_image=raster, clip_gpkg=local_gpkg)
-                if debug:
-                    logging.debug(f'Unique values in loaded raster: {np.unique(img_array)}\n'
-                                  f'Shape of raster: {img_array.shape}')
+        for info in tqdm(list_img, desc='Inferring from images', position=0, leave=True):
+            with start_run(run_name=Path(info['tif']).name, nested=True):
```

Comment:
There's a little bug here. at line 475, start_run is imported only if a mlflow uri is inputted. We can just import everything at the top of the script no matter if the uri is given or not. I can address this bugfix in my next PR.